lib/active_merchant/billing/gateways/stripe.rb

```diff
@@ -694,6 +695,10 @@ def api_request(method, endpoint, parameters = nil, options = {})
 
       def commit(method, url, parameters = nil, options = {})
         add_expand_parameters(parameters, options) if parameters
+
+        return Response.new(false, 'Invalid API Key provided') if test? && !key(options).include?('sk_test')
+        return Response.new(false, 'Invalid API Key provided') if !test? && !key(options).include?('sk_live')
```

Comment:
Would it be more accurate and maybe more secure to use `.starts_with?` instead of `.include?` here?